app/controlplane/plugins/core/webhook/v1/webhook.go

```diff
@@ -0,0 +1,345 @@
+}
+
+// New initializes the webhook integration
+func New(l log.Logger) (sdk.FanOut, error) {
+	base, err := sdk.NewFanOut(
+		&sdk.NewParams{
+			ID:          "webhook",
+			Version:     "1.0",
+			Description: "Send Attestation and SBOMs to a generic webhook URL as JSON payloads",
+			Logger:      l,
+			InputSchema: &sdk.InputSchema{
+				Registration: registrationRequest{},
+				Attachment:   attachmentRequest{},
+			},
+		},
+		// Subscribe to SBOMs only, attestations are handled separately
+		sdk.WithInputMaterial(schemaapi.CraftingSchema_Material_SBOM_CYCLONEDX_JSON),
+		sdk.WithInputMaterial(schemaapi.CraftingSchema_Material_SBOM_SPDX_JSON),
+	)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create FanOut integration: %w", err)
+	}
+
+	return &Integration{
+		FanOutIntegration: base,
+		client:            &http.Client{},
+	}, nil
+}
+
+// Register is executed when registering the webhook integration
+func (i *Integration) Register(ctx context.Context, req *sdk.RegistrationRequest) (*sdk.RegistrationResponse, error) {
+	i.Logger.Info("registration requested")
+
+	// Parse the registration payload
+	var regReq registrationRequest
+	if err := sdk.FromConfig(req.Payload, &regReq); err != nil {
+		i.Logger.Errorw("failed to parse registration payload", "error", err)
+		return nil, fmt.Errorf("invalid registration request: %w", err)
+	}
+
+	// Validate the URL
+	if err := validateURL(regReq.URL); err != nil {
+		i.Logger.Errorw("invalid webhook URL", "error", err, "url", regReq.URL)
+		return nil, fmt.Errorf("invalid webhook URL: %w", err)
+	}
+
+	// Optionally, perform a test request to ensure the webhook URL is reachable
+	if err := testWebhookURL(i.client, regReq.URL); err != nil {
+		i.Logger.Errorw("unable to reach webhook URL", "error", err, "url", regReq.URL)
+		return nil, fmt.Errorf("unable to reach webhook URL: %w", err)
+	}
+
+	// Store the URL in credentials
+	credentials := &sdk.Credentials{
+		URL: regReq.URL, // Storing the URL in the URL field
+	}
+
+	// No additional state needed
+	rawConfig, err := sdk.ToConfig(&registrationState{})
+	if err != nil {
+		i.Logger.Errorw("failed to marshal registration state", "error", err)
+		return nil, fmt.Errorf("marshalling configuration: %w", err)
+	}
+
+	return &sdk.RegistrationResponse{
+		Credentials:   credentials,
+		Configuration: rawConfig,
+	}, nil
+}
+
+// Attach is executed when attaching the webhook integration to a workflow
+func (i *Integration) Attach(_ context.Context, req *sdk.AttachmentRequest) (*sdk.AttachmentResponse, error) {
+	// Parse the attachment payload
+	var attachReq attachmentRequest
+	if err := sdk.FromConfig(req.Payload, &attachReq); err != nil {
+		i.Logger.Errorw("failed to parse attachment payload", "error", err)
+		return nil, fmt.Errorf("invalid attachment request: %w", err)
+	}
+
+	// Split the materials string into a slice
+	materials := []string{}
+	if attachReq.Materials != "" {
```

Comment:
as I mentioned in another thread, I think it would be best to limit these settings to explicit `send-sbom` and `send-attestation` options.

 Otherwise a) we'll give the false sense that this plugin supports more than just sboms, and b) it's hard for the user to know what values are valid as materials, is it `CYCLONEDX_SBOM` or `sbom`?